tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/NonParticipatingTckResource.java

```diff
@@ -0,0 +1,154 @@
+    @LRA(value = LRA.Type.NEVER)
+    public Response neverRunWithLRA(@HeaderParam(LRA_HTTP_HEADER) String lraId) {
+        return Response.ok(lraId).build();
+    }
+
+    @PUT
+    @Path(NonParticipatingTckResource.START_AND_END_PATH)
+    @LRA(value = LRA.Type.REQUIRES_NEW,
+            cancelOnFamily = Response.Status.Family.SERVER_ERROR) // default is to end the LRA
+    public Response startAndEndLRA(@HeaderParam(LRA_HTTP_HEADER) String lraId,
+                                   @DefaultValue("200") @QueryParam(STAUS_CODE_QUERY_NAME) int coerceStatus) {
+        return Response.status(coerceStatus).entity(checkLRANotNull(lraId)).build();
+    }
+
+    @PUT
+    @Path(NonParticipatingTckResource.START_BUT_DONT_END_PATH)
+    @LRA(value = LRA.Type.REQUIRES_NEW, end = false,
+            cancelOnFamily = Response.Status.Family.SERVER_ERROR)
+    public Response startDontEndLRA(@HeaderParam(LRA_HTTP_HEADER) String lraId,
+                                    @DefaultValue("200") @QueryParam(STAUS_CODE_QUERY_NAME) int coerceStatus) {
+        return Response.status(coerceStatus).entity(checkLRANotNull(lraId)).build();
+    }
+
+    @PUT
+    @Path(NonParticipatingTckResource.END_PATH)
+    @LRA(value = LRA.Type.MANDATORY,
+            cancelOnFamily = Response.Status.Family.SERVER_ERROR) // default is to end the LRA
+    public Response endLRA(@HeaderParam(LRA_HTTP_HEADER) String lraId,
+                           @DefaultValue("200") @QueryParam(STAUS_CODE_QUERY_NAME) int coerceStatus) {
+        return Response.status(coerceStatus).entity(checkLRANotNull(lraId)).build();
+    }
+
+    @PUT
+    @Path(NonParticipatingTckResource.START_AND_END_NESTED_PATH)
+    @LRA(value = LRA.Type.MANDATORY,
+            cancelOnFamily = Response.Status.Family.SERVER_ERROR) // default is to end the LRA
+    @NestedLRA
+    public Response startAndEndNestedLRA(@HeaderParam(LRA_HTTP_HEADER) String lraId) {
+        return Response.ok(lraId).build();
+    }
+
+    @PUT
+    @Path(NonParticipatingTckResource.START_BUT_DONT_END_NESTED_PATH)
+    @LRA(value = LRA.Type.MANDATORY, end = false,
+            cancelOnFamily = Response.Status.Family.SERVER_ERROR)
+    @NestedLRA
+    public Response startAndDontEndNestedLRA(@HeaderParam(LRA_HTTP_HEADER) String lraId) {
+        return Response.ok(lraId).build();
+    }
+
+    @PUT
+    @Path(NonParticipatingTckResource.START_LRA_VIA_REMOTE_INVOCATIOM)
+    @LRA(value = LRA.Type.SUPPORTS, end = false,
+            cancelOnFamily = Response.Status.Family.SERVER_ERROR)
+    public Response notSupportedButCallServiceWhichStartsButDoesntEndAnLRA(@HeaderParam(LRA_HTTP_HEADER) String lraId) {
+        return Response.ok(invokeRestEndpoint(lraId, TCK_NON_PARTICIPANT_RESOURCE_PATH, START_BUT_DONT_END_PATH,
+                200)).build();
+    }
+
+    private String invokeRestEndpoint(String lra, String basePath, String path, int coerceResponse) {
+        Client client = ClientBuilder.newClient();
```

Comment:
The request target is initialised by a request scoped URIInfo bean so I would probably have to make the whole resource request scoped too (instead of ApplicationScoped) so you would still suffer a performance penalty.

Doing it the way I have done it is just easier to understand and reason about.